DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerPs.py

```diff
@@ -775,51 +775,51 @@ def _setFileParameter(self, fileID, paramName, paramValue, connection=False):
   # _getFileReplicas related methods
   #
 
-  def _getFileReplicas(self, fileIDs, fields_input=['PFN'], allStatus=False, connection=False):
+  def _getFileReplicas(self, fileIDs, fields_input=None, allStatus=False, connection=False):
     """ Get replicas for the given list of files specified by their fileIDs
         :param fileIDs : list of file ids
-        :param fields_input : metadata of the Replicas we are interested in
+        :param fields_input : metadata of the Replicas we are interested in (default to PFN)
         :param allStatus : if True, all the Replica statuses will be considered,
                            otherwise, only the db.visibleReplicaStatus
 
         :returns S_OK with a dict { fileID : { SE name : dict of metadata } }
     """
 
-    connection = self._getConnection(connection)
+    if fields_input is None:
+      fields_input = ['PFN']
 
     fields = list(fields_input)
 
+    # always add Status in the list of required fields
     if 'Status' not in fields:
       fields.append('Status')
 
-    replicas = {}
+    # We initialize the dictionnary with empty dict
+    # as default value, because this is what we want for
+    # non existing replicas
+    replicas = {fileID: {} for fileID in fileIDs}
 
     # Format the status to be used in a IN clause in the stored procedure
     fStatus = stringListToString(self.db.visibleReplicaStatus)
 
     fieldNames = ["FileID", "SE", "Status", "RepType", "CreationDate", "ModificationDate", "PFN"]
 
-    for fileID in fileIDs:
-
-      result = self.db.executeStoredProcedureWithCursor('ps_get_all_info_of_replicas',
-                                                        (fileID, allStatus, fStatus))
+    for chunks in breakListIntoChunks(fileIDs, 1000):
+      # Format the FileIDs to be used in a IN clause in the stored procedure
+      formatedFileIds = intListToString(chunks)
+      result = self.db.executeStoredProcedureWithCursor('ps_get_all_info_of_replicas_bulk',
```

Comment:
Changes to the DBs should be done only for new releases. Here you end up imposing the deploy of a new stored procedure in patch 40 of v7r1. It's not correct.

My suggestion is that here you try the `bulk` procedure and fall back to the old one if not present. Then we repeat in the v7r2 notes about deploying the new `bulk` procedure.

Reply:
In general I totally agree. However this is for the LHCb DFC that I think absolutely nobody but use uses... I'll have a quick look if I can do that in a neat way but I wouldn't bother too much... I may as well move all these managers in LHCbDIRAC